tools/LogCollector/source/omiHighCPUDiagnostics.sh

```diff
@@ -0,0 +1,207 @@
+        split -l $omiagent_log_line $omiagent_log_file omiserversnap
+        find $omi_log_path -maxdepth 1 -iname 'omiagentsnap*' -not -name 'omiagentsnapaa' -exec cat {} +>>$omiagent_trace_path
+        rm -f $(omi_log_path)/omiagentsnap*
+        omiagent_log_line=$omiagent_log_line1
+    fi
+}
+
+dump_scx_omi_log_diff_tail()
+{
+    echo -e "\n\n" >>$omiagent_trace_path
+    echo "OMI Server log Diff in last 10 min">>$omiagent_trace_path
+    echo "------------------------------------------------">>$omiagent_trace_path
+    omiserver_log_line1=`wc -l $omi_log_file |awk '{print $1}'`
+    if [ $omiserver_log_line1 -ne $omiserver_log_line ]; then
+        lineNo=`expr $omiserver_log_line1 - $omiserver_log_line`
+        tail -n $lineNo $omi_log_file >>$omiagent_trace_path
+        omiserver_log_line=$omiserver_log_line1
+    fi
+
+    echo -e "\n\n" >>$omiagent_trace_path
+    echo "OMIagent  log Diff in last 10 min">>$omiagent_trace_path
+    echo "------------------------------------------------">>$omiagent_trace_path
+    scx_log_line1=`wc -l $scx_log_file |awk '{print $1}'`
+    if [ $scx_log_line1 -ne $scx_log_line ]; then
+        lineNo=`expr $scx_log_line1 - $scx_log_line`
+        tail -n $lineNo $scx_log_file>>$omiagent_trace_path
+        scx_log_line=$scx_log_line1
+    fi
+
+
+    echo -e "\n\n" >>$omiagent_trace_path
+    echo "SCX log Diff in last 10 min">>$omiagent_trace_path
+    echo "------------------------------------------------">>$omiagent_trace_path
+    omiagent_log_line1=`wc -l $omiagent_log_file |awk '{print $1}'`
+    if [ $omiagent_log_line1 -ne $omiagent_log_line ]; then
+        lineNo=`expr $omiagent_log_line1 - $omiagent_log_line`
+        tail -n $lineNo $omiagent_log_file>>$omiagent_trace_path
+        omiagent_log_line=$omiagent_log_line1
+    fi
+}
+
+
+while [ $count -gt 0 ]
+do
+	date>>$omiagent_trace_path
+    ppuc=(`top -b -n1|grep omiagent|tee -a $omiagent_trace_path|awk '{print $9;print $1}'`)
+
+	index=0
+	index1=0;
+	while [ $index -lt ${#ppuc[@]} ]
+	do
+ 		if [ $(echo "${ppuc[$index]} > $cpu_threshold"|bc) -eq 1 ]; then
+            ind=`expr $index + 1`
+            echo -e "\n\n" >>$omiagent_trace_path
+            echo "Threads with High CPU Utilization:">>$omiagent_trace_path
+            echo "------------------------------------------------">>$omiagent_trace_path
+            pt=(`top -b -H -p ${ppuc[$ind]} -n1|grep omiagent|tee -a $omiagent_trace_path|awk '{print $9;print $1}'`)
+
+            echo -e "\n\n" >>$omiagent_trace_path
+            echo "Stacktrace:">>$omiagent_trace_path
+            echo "----------------------------------------------">>$omiagent_trace_path
+			echo -e "\n\n" >>$omiagent_trace_path
+            echo "Stacktrace for Process: ${ppuc[$ind]}">>$omiagent_trace_path
+            echo "----------------------------------------------">>$omiagent_trace_path
+            sudo gdb -p ${ppuc[$ind]} -batch -ex "thread apply all bt" -ex quit>>$omiagent_trace_path
```

Comment:
sudo: gdb: command not found
ubuntu 18

Reply:
possibly related
41	../sysdeps/unix/sysv/linux/select.c: No such file or directory.

Reply:
done